ui/src/message_popup/renderers/read_more.jsx

```diff
@@ -0,0 +1,64 @@
+/* @flow weak */
+import React from 'react';
+
+/*
+  Component that displays a text block that can be expanded or collapsed.
+*/
+export default React.createClass({
+  displayName: 'ReadMore',
+
+  propTypes: {
+    fulltext: React.PropTypes.node.isRequired,
+    charCount: React.PropTypes.node,
+  },
+
+  getInitialState() {
+    return {
+      expanded: false
+    };
+  },
+
+  expand() {
+    this.setState({
+      expanded: true
+    });
+  },
+
+  collapse() {
+    this.setState({
+      expanded: false
+    });
+  },
+
+  render() {
+    var charCount = (this.props.charCount === undefined) ? 140 : this.props.charCount;
```

Comment:
You can use the React method `getDefaultProps` to express default values like this.